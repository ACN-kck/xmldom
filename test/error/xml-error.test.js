'use strict'

const { getTestParser } = require('../get-test-parser')

describe('errorHandle', () => {
	it.each(['text/html', 'text/xml'])('unclosed document in %s', (mimeType) => {
		const { errors, parser } = getTestParser()

		const actual = parser.parseFromString('<img>', mimeType).toString()

		expect({ actual, ...errors }).toMatchSnapshot()
	})

	it.each([
		['<test><!--', '<test/>'],
		['<r', '<r/>'],
	])('invalid xml node "%s"', (input, expected) => {
		const { errors, parser } = getTestParser()

		const actual = parser
			.parseFromString(input, 'text/xml')
			.documentElement.toString()

		expect({ actual, ...errors }).toMatchSnapshot({ actual: expected })
	})

	it('html attribute (miss quote)', () => {
		const { errors, parser } = getTestParser()

		const actual = parser
			.parseFromString('<img attr=1/>', 'text/html')
			.toString()

		expect({ actual, ...errors }).toMatchSnapshot()
	})

	it('valid html attribute value (<>&)', () => {
		const { errors, parser } = getTestParser()

		const actual = parser
			.parseFromString('<img attr="<>&"/>', 'text/html')
			.toString()

		expect({ actual, ...errors }).toMatchSnapshot()
	})
})
