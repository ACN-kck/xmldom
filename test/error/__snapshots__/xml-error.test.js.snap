// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorHandle html attribute (miss quote) 1`] = `
Object {
  "actual": "<img attr=\\"1\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"/>",
  "warning": Array [
    "[xmldom warning]	attribute \\"1\\" missed quot(\\")!!
@#[line:1,col:1]",
  ],
}
`;

exports[`errorHandle invalid xml node "<r" 1`] = `
Object {
  "actual": "<r/>",
  "error": Array [
    "[xmldom error]	unexpected end of input
@#[line:1,col:1]",
  ],
  "warning": Array [
    "[xmldom warning]	unclosed xml attribute
@#[line:1,col:1]",
  ],
}
`;

exports[`errorHandle invalid xml node "<test><!--" 1`] = `
Object {
  "actual": "<test/>",
  "error": Array [
    "[xmldom error]	Unclosed comment
@#[line:1,col:7]",
  ],
  "warning": Array [
    "[xmldom warning]	unclosed xml attribute
@#[line:1,col:1]",
  ],
}
`;

exports[`errorHandle unclosed document in text/xml 1`] = `
Object {
  "actual": "<img/>",
  "warning": Array [
    "[xmldom warning]	unclosed xml attribute
@#[line:1,col:1]",
  ],
}
`;

exports[`errorHandle unclosed document in text/html 1`] = `
Object {
  "actual": "<img xmlns=\\"http://www.w3.org/1999/xhtml\\"/>",
}
`;

exports[`errorHandle valid html attribute value (<>&) 1`] = `
Object {
  "actual": "<img attr=\\"&lt;>&amp;\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"/>",
}
`;
